kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/RecordEncryptionFilter.java

```diff
@@ -57,6 +61,27 @@ public class RecordEncryptionFilter<K>
         this.encryptionManager = encryptionManager;
         this.decryptionManager = decryptionManager;
         this.filterThreadExecutor = filterThreadExecutor;
+        checkCipherSuite();
+    }
+
+    private static void checkCipherSuite() {
+        checkCipherSuite(CipherSpec::newCipher);
+    }
+
+    /* exposed for testing */ static void checkCipherSuite(Function<CipherSpec, Cipher> cipherFunc) {
+        List<CipherSpec> failures = Arrays.stream(CipherSpec.values()).flatMap(cipherSpec -> {
+            try {
+                cipherFunc.apply(cipherSpec);
+                return Stream.empty();
+            }
+            catch (Exception e) {
+                log.error("A Cipher could not be constructed for CipherSpec {}", cipherSpec, e);
+                return Stream.of(cipherSpec);
+            }
+        }).toList();
+        if (!failures.isEmpty()) {
+            throw new EncryptionConfigurationException("Cipher Suite check failed, one or more ciphers could not be loaded: " + failures);
```

Comment:
It might be worth having the `java.*` system properties logged on startup so there is a record of the platform the the user is using in the logs.  That'd provide the information needed to diagnose this situation in one place.

Reply:
Implemented in #1066

Reply:
Ohh I'd miss read this comment when scanning emails, and thought it was requesting a dump of all system props not just the vendor et al.

Reply:
Though given the number of system properties related to encryption and ciphers mentioned in the [configure crypto](https://www.java.com/en/configure_crypto.html) guide I wonder if should just dump the lot? Or maybe we should log: 

- jdk.tls.client.cipherSuites, 
- jdk.tls.server.cipherSuites
- https.cipherSuites
- jdk.tls.namedGroups
- jdk.tls.disabledAlgorithms

And probably just about anything else under `jdk.tls`

Reply:
Those are properties used for TLS, so I don't think they are relevant to this PR.  I think we can defer thinking about the logging of those those until #1006